### PR TITLE
Create in-place variants for encrypt_auth/decrypt_auth

### PR DESCRIPTION
**Pull request by kedars**
This is invaluable for embedded systems

### CODE REVIEW

**Comment by kedars** [comment]
ping... any review comments for including this?

**Comment by jethrogb** [comment]
I think it's a bit weird to have the in-place version for the single-step authenticated crypto functions because input and output sizes don't match. I'd be supportive of an in-place version of `update`.

**Comment by kedars** [comment]
How about splitting the tag into a separate out parameter for the in-place variant? The plain/cipher sizes would then match, and the caller gets the control

**Comment by jethrogb** [comment]
I'm not sure the MbedTLS API supports that?

**Comment by kedars** [comment]
Took a stab at it, let me know what you think

**Comment by kedars** [comment]
For the `update` operation, `mbedtls_cipher_update()`, the [mbedTLS documentation](https://tls.mbed.org/api/cipher_8h.html#a959598dfef9d7f2add8861e0c518f303) specifically forbids the output and input buffers to be the same.

File: mbedtls/src/cipher/mod.rs [hunk 1]
```diff
@@ -401,6 +427,44 @@ fn ccm() {
...
+    let cipher = Cipher::<_, Authenticated, _>::new(
+        raw::CipherId::Aes,
+        raw::CipherMode::CCM,
+        (k.len() * 8) as _,
+    )
+    .unwrap();
+    let cipher = cipher.set_key_iv(&k, &iv).unwrap();
+    let (data, tag) = c.split_at_mut(4);
+    cipher.decrypt_auth_inplace(&ad, data, tag).unwrap();
+    assert_eq!(validate_plain, c[0..4]);
```
Comment (jethrogb) [inline mbedtls/src/cipher/mod.rs:465]:
```suggestion
    assert_eq!(validate_plain, data);
```

**Review by jethrogb — COMMENTED** [review]
Looks good now! One small nit.

File: mbedtls/src/cipher/mod.rs (hunk 1 above)
Comment (kedars) [reply mbedtls/src/cipher/mod.rs]: makes sense, done 👍

**Pull request merged**